main.go

```diff
@@ -83,11 +97,22 @@ func main() {
 
 	bootstrapCount := 0
 
+	var parentBlockHash *common.Hash
 	for {
-		block, err := client.BlockByNumber(context.Background(), blockNum)
+		var block *types.Block
+		if parentBlockHash == nil {
+			block, err = client.BlockByNumber(context.Background(), blockNum)
+		} else {
+			block, err = client.BlockByHash(context.Background(), *parentBlockHash)
+		}
 		if err != nil {
 			log.Fatal(err)
 		}
+		if block == nil {
+			break
```

Comment:
would it be better to panic here?    This shouldn't happen unless you're going back before block 0 I think?